website/blog/2020-06-27-lithium.md

```diff
@@ -0,0 +1,503 @@
+
+The support includes most of the LSP features such as diagnostics, completions,
+references, go to definition, but does not include running or debugging the
+scripts. This new feature works by starting an additional Ammonite BSP server
+that compiles the scripts and provides additional information about them such as
+scala version or information about the added dependencies. You can change the
+default JVM parameters for the server such as `-Xmx`, which will limit the used
+memory in Metals settings.
+
+Each of the scripts needs to be first imported into Ammonite BSP server, which
+can be done manually by clicking on the pop up whenever new script is opened or
+selecting automatic import at the start, which will import any open scripts.
+
+It's important to note that this is an early approach to the support, which
+might still require some improvements both on the Metals and Ammonite side. You
+learn more about the new feature or comment the in the
+[blogpost](https://contributors.scala-lang.org/t/ammonite-support-in-metals/4306)
+on the Scala contributors forum.
+
+Most notable features of Ammonite scripts are:
+
+- importing other scripts, Metals provide completions to select existing files
+  based on the filesystem information
+
+```scala
+//main.sc
+import $file.hello-world
+val greeting = hello-world.greeting
+
+//hello-world.sc
+val greeting = "Hello world!"
+```
+
+- importing dependecies
+
+```scala
+import $ivy.`com.lihaoyi::scalatags:0.7.0`
+import scalatags.Text.all._
+
+val rendered = div("Moo").render
+```
+
+## Worksheet dependency support
+
+In addition to Ammonite support, which will be useful to anyone interested in
+Scala scripting, we've also improved the support for Metals Scala worksheets.
+Thanks to the efforts of [olafurpg](https://github.com/olafurpg), we now are
+able to provide importing dependencies inside your Scala worksheets with
+completions and diagnostics working right from the bat.
+
+![worksheet-imports](https://i.imgur.com/qXgdNWM.gif)
+
+You can add them using the following syntax:
+
+```
+$dep.`organisation::artifact:version`
+```
+
+for example:
+
+```
+$dep.`com.lihaoyi::scalatags:0.9.0`
+```
+
+`::` is the same as `%%` in sbt, which will append the current scala binary
+version to the artifact name.
+
+## New project provider
+
+Creating new Scala projects is usually a bit of work to get started, which is a
+problem for both beginners and experts alike. Normally you don't want a lot of
+time to figure out the best setup for starting with a new technology or the
+language itself.
+
+There are great tools like [giter8](http://www.foundweekends.org/giter8/), which
+is used in the `sbt new` command, that enables people to use already existing
+Scala templates found on Github such as https://github.com/scala/scala-seed.g8.
+The problem is when you don't use sbt or you don't know the exact template you
+want to use.
+
+In this version we added an additional command called `New Scala Project` that
```

Comment:
```suggestion
To address this, we've added an additional command called `New Scala Project` that
```

Reply:
I think since we outlined the problem in the sentence before we can change this a bit to draw a line from `existing problem` ----> `our solution`.